src/pages/AdminEditUser.js

```diff
@@ -0,0 +1,303 @@
+import _ from 'lodash';
+import {union, contains, map, isEmpty} from 'lodash/fp';
+import React, {Component} from 'react';
+import {button, div, form, hh, input, label} from 'react-hyperscript-helpers';
+import {User} from '../libs/ajax';
+import {USER_ROLES} from '../libs/utils';
+import {ResearcherReview} from './ResearcherReview';
+import editUserIcon from '../images/icon_edit_user.png';
+import {PageHeading} from "../components/PageHeading";
+
+const adminRole = {'roleId': 4, 'name': USER_ROLES.admin};
+const researcherRole = {'roleId': 5, 'name': USER_ROLES.researcher};
+const signingOfficialRole = {'roleId': 7, 'name': USER_ROLES.signingOfficial};
+
+export const AdminEditUser = hh(class AdminEditUser extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {},
+      displayName: '',
+      email: '',
+      displayNameValid: false,
+      updatedRoles: [researcherRole],
+      emailPreference: false
+    };
+
+    this.nameRef = React.createRef();
+  }
+
+  async componentDidMount() {
+    const user = await User.getById(this.props.match.params.dacUserId);
+    const currentRoles = _.map(user.roles, (ur) => {
+      return {'roleId': ur.roleId, 'name': ur.name};
+    });
+    const updatedRoles = isEmpty(currentRoles) ? [researcherRole] : currentRoles;
+    this.setState({
+      displayName: user.displayName,
+      email: user.email,
+      user: user,
+      updatedRoles: updatedRoles,
+      emailPreference: user.emailPreference
+    },
+    () => {
+      this.setState({
+        displayNameValid: this.nameRef.current.validity.valid,
+      });
+    });
+  }
+
+  OKHandler = async (event) => {
+    event.persist();
+    event.preventDefault();
```

Comment:
You'll need to keep this line in because the input/submit fields are in a form tag. Since the default browser behavior is to refresh the page on a form submit, e.preventDefault() stops that from happening.